### PR TITLE
Calculation manager

### PR DESCRIPTION
**Pull request by GloeStoe**
added a calculation manager with multiprocessing and fixed inconsistencies in the calculation phases

### CODE REVIEW

File: src/osm_configurator/model/project/calculation/calculation_manager.py [hunk 1]
```diff
@@ -10,39 +28,55 @@ class CalculationManager:
...
         """
         Gets called when we first create an object of this class. It saves all information it needs for
         managing the calculations.
 
         Args:
             configuration_manager (configuration_manager.ConfigurationManager): Saves all information required to configure the calculation.
         """
-        pass
+        self._config_manager = configuration_manager
+        self._phases: List[ICalculationPhase] = [
```
Comment (LuposX) [inline src/osm_configurator/model/project/calculation/calculation_manager.py:40]: this is hardcoded, bad if you want to add a new phase, only do this if there is no other way.
Maybe more as a constant? idk

File: src/osm_configurator/model/project/calculation/calculation_manager.py [hunk 2]
```diff
@@ -10,39 +28,55 @@ class CalculationManager:
...
+            geo_data_phase.GeoDataPhase(),
+            tag_filter_phase.TagFilterPhase(),
+            reduction_phase.ReductionPhase(),
+            attractivity_phase.AttractivityPhase(),
+            aggregation_phase.AggregationPhase()
+        ]
+        self._calculation_state = (calculation_state_enum.CalculationState.NOT_STARTED_YET,
```
Comment (LuposX) [inline src/osm_configurator/model/project/calculation/calculation_manager.py:47]: typehint

File: src/osm_configurator/model/project/calculation/calculation_manager.py [hunk 3]
```diff
@@ -51,6 +85,30 @@ def start_calculation(self, starting_point):
             starting_point (calculation_phase_enum.CalculationPhase): The starting phase of the calculation. The calculations start from this phase.
 
         Returns:
-            calculation_phase_enum.CalculationState: The state of the calculation, after trying to start the calculations.
+            calculation_phase_enum.CalculationState, str: The state of the calculation, after trying to start the calculations.
         """
-        pass
+        if len([x for x in self._phases if x.get_calculation_phase_enum() == starting_point]) != 1:
+            raise ValueError("There is not a unique ICalculationPhase with the given staring point")
```
Comment (LuposX) [inline src/osm_configurator/model/project/calculation/calculation_manager.py:91]: maybe use better exception, idk

**Review by LuposX — COMMENTED** [review]
In addition to what I have written, i do miss developer comments, some developer comments in your code would tremendously help if someone want to change your code.

**Review by LuposX — APPROVED** [review]

File: src/osm_configurator/model/project/calculation/calculation_manager.py (hunk 2 above)
Comment (GloeStoe) [reply src/osm_configurator/model/project/calculation/calculation_manager.py]: done

File: src/osm_configurator/model/project/calculation/calculation_manager.py (hunk 3 above)
Comment (GloeStoe) [reply src/osm_configurator/model/project/calculation/calculation_manager.py:91]: idk a better one

File: src/osm_configurator/model/project/calculation/calculation_manager.py (hunk 1 above)
Comment (GloeStoe) [reply src/osm_configurator/model/project/calculation/calculation_manager.py:40]: denied change

**Pull request merged**